tests/rptest/services/flink.py

```diff
@@ -0,0 +1,405 @@
+            idx = line.find("JobID")
+            if idx > 0:
+                id = line[idx + 6:]
+                self.logger.debug(f"Found JobID:\n{id}")
+                generated_ids.append(id)
+
+        return generated_ids
+
+    def _get(self, rest_handle):
+        """
+            Perform a GET to Flink REST API
+            More here:
+            https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/ops/rest_api/
+        """
+        url = f"{self._baseurl}{rest_handle}"
+        try:
+            r = requests.get(url)
+        except Exception as e:
+            raise RuntimeError(f"Failed to get data from '{url}'") from e
+        return r.json()
+
+    def list_jobs(self):
+        """
+            List all jobs
+        """
+        handle = "/jobs"
+        return self._get(handle)
+
+    def get_job_by_id(self, jobid):
+        """
+            Gets job specs
+        """
+        handle = f"/jobs/{jobid}"
+        return self._get(handle)
+
+    def start_node(self, node):
+        # Prepare config
+        # Dump yaml and prevent java options to be divided
+        conf = yaml.dump(flink_config, width=float("inf"))
+        self.logger.info("Creating flink configuration file")
+        node.account.create_file(self.FLINK_CONFIG_FILE_PATH, conf)
+
+        # Start
+        self.logger.info("Starting Flink standalone cluster")
+        out = node.account.ssh_output(self.FLINK_START)
+        self.logger.info(f"Response:\n{out.decode()}")
+
+        # Save node data for future log handling
+        self.node = node
+        self.hostname = node.account.hostname
+        return
+
+    def stop_node(self, node):
+        # Stop cluster
+        self.logger.info("Stopping flink standalone cluster")
+        out = node.account.ssh_output(self.FLINK_STOP)
+        self.logger.info(f"Response:\n{out.decode()}")
+
+        # No local node or hostname removing as they needed for log extraction
+        return
+
+    def get_active_jobs(self):
+        """
+            Get all active jobs from flink REST API
+        """
+        jobs = self.list_jobs()
+        active = [
+            job for job in jobs['jobs']
+            if job['status'] in self._active_statuses
+        ]
+        return active
+
+    def _has_active_jobs(self):
+        """
+            Get active job list, log aggregated status count
+            Return Bool valus if active jobs >0
+        """
+        jobs = self.get_active_jobs()
+        map = {}
+        for j in jobs:
+            s = j['status']
+            if s not in map:
+                map[s] = 1
+            else:
+                map[s] += 1
+        out = [f"{k}: {v}" for k, v in map.items()]
+        if len(out) > 0:
+            self.logger.debug(f"Flink active jobs status: {', '.join(out)}")
+        else:
+            self.logger.debug("Flink has no active jobs")
+
+        return len(jobs) > 0
+
+    def wait_node(self, node, timeout_sec=1800):
```

Comment:
nit: 1800 seems too high.

Reply:
Yes, I missed that. will update to 120. This one was used while debugging other threads